form-validation/src/Form.js

```diff
@@ -0,0 +1,177 @@
+import React from 'react';
+
+class Form extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = this.constructor.initialState;
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  static get initialState() {
+    return ({
+      fields: {
+        loginId: '',
+        mail: '',
+        userName: '',
+        timezone: '',
+        homepage: '',
+        aboutMe: '',
+        notification: '',
+      },
+      errors: {}
+    });
+  }
+
+  handleInputChange(event) {
+    const target = event.target;
+    const value = target.type === 'checkbox' ? target.checked : target.value;
+    const name = target.name;
+
+    this.setState((prevState) => ({
+      fields: {
+        ...prevState.fields,
+        [name]: value
+      }
+    }));
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    this.clearErrors();
+    if(this.validateFields(event)) {
+      this.resetState();
+    }
+  }
+
+  resetState() {
+    this.setState(this.constructor.initialState);
+  }
+
+  clearErrors() {
+    this.setState({
+      errors: {}
+    });
+  }
+
+  validateFields(event) {
+    let isValid = true
+    for(const element of event.target.elements) {
+      if(this.validatePresence(element)) {
+        isValid = false
+        this.addPresenceValidationError(element);
+      } else if(element.name === 'mail') {
+        isValid = this.validateEmailFormat(element);
+      } else if(element.name === 'homepage') {
+        isValid = this.validateURLFormat(element);
+      } else if(element.name === 'aboutMe') {
+        isValid = this.validateLength(element);
+      }
+    }
+    return isValid;
+  }
+
+  validatePresence(element) {
+    return (!element.value || (element.type === 'checkbox' && !element.checked) || element.selectedIndex <= 0);
+  }
+
+  addPresenceValidationError(element) {
+    this.setState((prevState) => ({
```

Comment:
You have set the error in 4 functions, you can create another method to set errors and just pass element.name & error message